azurelinuxagent/common/event.py

```diff
@@ -179,6 +181,53 @@ def _save(self):
     ]
 
 
+def parse_json_event(data_str):
```

Comment:
These were previously on the monitor handle. Note that I change the exception handling in parse_event.